app/models/alchemy/element.rb

```diff
@@ -313,17 +313,29 @@ def copy_nested_elements_to(target_element)
 
     private
 
-    # Returns previous or next public element from same page.
+    def select_element(elements, name, order)
+      elements = elements.named(name) if name.present?
+      elements.reorder(position: order).limit(1).first
+    end
+
+    # Returns the next public element from same page.
     #
     # @param [String]
-    #   Pass '>' or '<' to find next or previous public element.
+    #   Pass an element name to get next of this kind.
+    #
+    def next_element(name = nil)
+      elements = page.elements.published.where('position > ?', position)
```

Comment:
I think there is no need for this extra method. Just put these two lines into the `next` method, actually less LOC